sycl/include/syclcompat/launch_policy.hpp

```diff
@@ -192,6 +192,17 @@ launch_policy(dim3, dim3, Ts...) -> launch_policy<
     detail::has_type<local_mem_size, std::tuple<Ts...>>::value>;
 
 namespace detail {
+// Custom std::apply helpers to enable inlining
+template <class F, class Tuple, size_t... Is>
+__syclcompat_inline__ constexpr void apply_expand(F f, Tuple t,
+                                                  std::index_sequence<Is...>) {
+  [[clang::always_inline]] f(get<Is>(t)...);
+}
+
+template <class F, class Tuple>
+__syclcompat_inline__ constexpr void apply_helper(F f, Tuple t) {
+  apply_expand(f, t, std::make_index_sequence<std::tuple_size<Tuple>{}>{});
+}
```

Comment:
```suggestion
template <class F, class Tuple>
__syclcompat_inline__ constexpr void apply_helper(F&& f, Tuple&& t) {
  apply_expand(std::forward<F>(f), std::forward<Tuple>(t), std::make_index_sequence<std::tuple_size<Tuple>{}>{});
}
```

I think this is one of those cases that perfect forwarding makes sense. Same for the tuple in`apply_expand`.

Reply:
It is worth checking but that was part of the issue with `std::apply`. It uses `std::forward` which internally calls other functions that are not inlined. We can also try using `flatten` but I'm not sure if it will propagate the inlining all the way to the non-inlined calls.

Reply:
I guess if you are inline, they the effects of std::forwards are moot anyway.